Android/app/src/main/java/com/doubleslas/fifith/alcohol/ui/auth/LoginViewModel.kt

```diff
@@ -0,0 +1,38 @@
+package com.doubleslas.fifith.alcohol.ui.auth
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.map
+import com.doubleslas.fifith.alcohol.model.repository.AuthRepository
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.GoogleAuthProvider
+
+class LoginViewModel : ViewModel() {
+    private val authRepository by lazy { AuthRepository() }
+    private var firebaseAuth = FirebaseAuth.getInstance()
+
+    enum class AuthenticationState {
+        AUTHENTICATED, UNAUTHENTICATED, INVALID_AUTHENTICATION
+    }
+
+    val authenticationState = FirebaseUserLiveData().map { user ->
+        if (user != null) {
+            AuthenticationState.AUTHENTICATED
+        } else {
+            AuthenticationState.UNAUTHENTICATED
+        }
+    }
+
+    fun firebaseAuthWithGoogle(idToken: String) {
+        val credential = GoogleAuthProvider.getCredential(idToken, null)
+        firebaseAuth!!.signInWithCredential(credential)
+            .addOnCompleteListener {task ->
+                if (task.isSuccessful) {
+                    val user = firebaseAuth.currentUser?.displayName
+                    authRepository.completeLogin(credential)
+                    Log.d("junmin", user.toString())
+                }
+            }
```

Comment:
Firebase Login 은 model에서 처리할게요.
View 에서 구글로그인을 하면
ViewModel에선 해당 데이터에서 FIrebase Credential을 추출해서 Repository에게 넘기기만 하면 됩니다.

Reply:
@comforest 음.. 어떻게 하는걸까요?

Reply:
@junmin-Chang 
```suggestion
        authRepository.completeLogin(credential)
```
이렇게요 ㅎㅎ...

Reply:
@comforest 아 삭제된 부분을 호연님께서 모델에서 처리하신다는 말씀이었나요?

Reply:
넵 ㅎㅎ 모델 코드 보시면 있을거에요